airbyte-integrations/connectors/destination-s3-v2/src/test-integration/kotlin/io/airbyte/integrations/destination/s3_v2/S3V2WriteTest.kt

```diff
@@ -140,7 +141,7 @@ class S3V2WriteTestCsvGzip :
 class S3V2WriteTestAvroUncompressed :
     S3V2WriteTest(
         S3V2TestUtils.AVRO_UNCOMPRESSED_CONFIG_PATH,
-        TODO(),
+        NoopExpectedRecordMapper,
```

Comment:
It feels confusing that the stricter destinations use Noop, and the less strict use Uncoerced

Reply:
this is a thing I'll probably change for https://github.com/airbytehq/airbyte-internal-issues/issues/11218. I think this is only b/c the avro/parquet dumpers are implicitly coercing back to the expected schema - I want to switch that to be more like how iceberg does it, i.e.
* dumper maps naively from AvroValue -> AirbyteValue (so e.g. it would read timestamps as IntegerValue)
* expected record mapper maps expected record AirbyteValue (so e.g. convert TimestampTZValue -> IntegerValue)